src/pytesmo/time_series/anomaly.py

```diff
@@ -87,9 +87,24 @@ def calc_anomaly(Ser,
     return anomaly
 
 
+def _index_units(year, month, day, unit="day", respect_leap_years=True) -> (np.array, int):
+    """Provide the indices for the specified unit type and the index lenth"""
+    if respect_leap_years:
+        args = month, day, year
+    else:
+        args = month, day
+
+    idx_lut = {
+        "day": (doy(*args), 366),
+        "month": (month, 12),
+    }
+
+    return idx_lut[unit]
```

Comment:
Can you reformulate this as an if-else statement?
```
if unit == "day":
    return doy(*args), 366
elif unit == "month":
    return month, 12
else:
    raise ValueError(f"Invalid unit: {unit}")
```

This way the function `doy` is only called if it is really needed.